internal/errors/match.go

```diff
@@ -0,0 +1,99 @@
+package errors
+
+// Template is useful constructing Match Err templates.  Templates allow you to
+// match Errs without specifying a Code.  In other words, just Match using the
+// Errs: Kind, Op, etc.
+type Template struct {
+	Err       // Err embedded to support matching Errs
+	Kind Kind // Kind allows explicit matching on a Template without a Code.
+}
+
+// T creates a new Template for matching Errs.  Invalid parameters are ignored.
+// If more than is one parameter for a given type, only the last one is used.
+func T(args ...interface{}) *Template {
+	t := &Template{}
+	for _, a := range args {
+		switch arg := a.(type) {
+		case Code:
+			t.Code = arg
+		case string:
+			t.Msg = arg
+		case Op:
+			t.Op = arg
+		case *Err: // order is important, this match must before "case error:"
+			c := *arg
+			t.Wrapped = &c
+		case error:
+			t.Wrapped = arg
+		case Kind:
+			t.Kind = arg
+		default:
+			// ignore it
+		}
+	}
+	return t
+}
+
+// Info about the Template, which is useful when matching a Template's Kind with
+// an Err's Kind.
+func (t *Template) Info() Info {
+	if t == nil {
+		return errorCodeInfo[Unknown]
+	}
+	switch {
+	case t == nil:
```

Comment:
This case has already been checked in the `if t == nil` above, we should probably drop the `case` or the `if`

Reply:
ty!